views/xTree/XTreeView.js

```diff
@@ -0,0 +1,1323 @@
+
+        this.generation++;
+        if (this.generation <= this.maxGeneration) {
+            this.displayGeneration(nextPeople);
+        } else {
+            wtViewRegistry.clearStatus();
+        }
+    }
+
+    // Grab a parent profile for the next generation. If we don't have an id, we use a place-holder instead, so the
+    // display keeps going with an "Unknown" relative displayed.
+    async nextPerson(id) {
+        if (id > 0) {
+            // We used to get each profile individually.
+            // let data = await WikiTreeAPI.postToAPI({appId: XTreeView.APP_ID, action: "getPerson", key: id, fields: this.profileFields });
+            // if (data.length != 1) {
+            //     return this.blankPerson;
+            // } else {
+            //     return data[0].person;
+            // }
+
+            // Now, we gathered all of our ancestors in one swoop at init and stored them by id in this.ancestors.
+            if (this.ancestors[id] !== undefined) {
+                return this.ancestors[id];
+            } else {
+                return this.blankPerson;
+            }
+        }
+        return this.blankPerson;
+    }
+
+    displayPersonName(person) {
+        let html = "";
+
+        if (person.Id == 0) {
+            html += `[${this.unknownName()} Unknown]`;
+        } else {
+            if (!person.MiddleName) {
+                person.MiddleName = "";
+            }
+
+            html += "<b>";
+            html += `${person.FirstName} ${person.MiddleName} `;
+            if (person.Nicknames) {
+                html += `"${person.Nicknames}" `;
+            }
+            if (person.LastNameCurrent != person.LastNameAtBirth) {
+                html += ` (${person.LastNameAtBirth}) `;
+            }
+            html += `${person.LastNameCurrent}`;
+            if (person.Suffix) {
+                html += ` ${person.Suffix}`;
+            }
+        }
+
+        return html;
+    }
+
+    // This code takes the WikiTree API person data and renders it into HTML for appending to the view display container.
+    displayPerson(person) {
+        let html = `<p class="ahnentafelPerson" id="person_${this.ahnentafelNumber}">${this.ahnentafelNumber}. `;
+
+        if (person.Id == 0) {
+            html += `[${this.unknownName()} Unknown]`;
+        } else {
+            if (!person.MiddleName) {
+                person.MiddleName = "";
+            }
+            if (this.generation == 1) {
+                html += "<b>";
+                html += `${person.FirstName} ${person.MiddleName} `;
+                if (person.Nicknames) {
+                    html += `"${person.Nicknames}" `;
+                }
+                if (person.LastNameCurrent != person.LastNameAtBirth) {
+                    html += ` (${person.LastNameAtBirth}) `;
+                }
+                html += `${person.LastNameCurrent}`;
+                if (person.Suffix) {
+                    html += ` ${person.Suffix}`;
+                }
```

Comment:
btw, you can use your function `displayPersonName` (also in `else` branch of this `if` statement) - it seems very similar